author: Matthias Fuchs <mat69@gmx.net>
commit: r1227471

Further addresses CVE-2010-1000.
The file name of Metalink File is checked a better way, making it work under more conditions.

Index: kdenetwork-4.6.2/kget/ui/metalinkcreator/metalinker.cpp
===================================================================
--- kdenetwork-4.6.2.orig/kget/ui/metalinkcreator/metalinker.cpp	2011-01-19 23:21:04.000000000 +0100
+++ kdenetwork-4.6.2/kget/ui/metalinkcreator/metalinker.cpp	2011-04-10 12:08:49.650167078 +0200
@@ -583,7 +583,13 @@
         return false;
     }
 
-    if (name.contains(QRegExp("$(\\.\\.?)?/")) || name.contains("/../") || name.endsWith("/..")) {
+    if (name.endsWith('/')) {
+        kError(5001) << "Name attribute of Metalink::File does not contain a file name:" << name;
+        return false;
+    }
+
+    const QStringList components = name.split('/');
+    if (name.startsWith('/') || components.contains("..") || components.contains(".")) {
         kError(5001) << "Name attribute of Metalink::File contains directory traversal directives:" << name;
         return false;
     }
